lib/assessment.js

```diff
@@ -29,7 +45,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  array2 = [];
+  for (var i = 0; i < defaultsFunction.length; i++){
+    array2 = array2[defaultValue]
```

Comment:
This is assigning the value stored in `array2[defaultValue]` into `array2`, causing us to lose the reference to the array in the very first iteration.

Also, there is no local `defaultValue`, so it's grabbing it from an outside scope. There is no accounting for what it might be.